src/providers/krb5/krb5_child_handler.c

```diff
@@ -418,6 +420,19 @@ errno_t set_extra_args(TALLOC_CTX *mem_ctx, struct krb5_ctx *krb5_ctx,
         c++;
     }
 
+    chain_id = sss_chain_id_get();
+    if (chain_id > 0) {
+        extra_args[c] = talloc_asprintf(extra_args,
+                                        "--"CHILD_OPT_CHAIN_ID"=%lu",
+                                        chain_id);
+        if (extra_args[c] == NULL) {
+            DEBUG(SSSDBG_OP_FAILURE, "talloc_asprintf failed.\n");
+            ret = ENOMEM;
+            goto done;
+        }
+    }
+    c++;
```

Comment:
Hi,

you should call `c++` inside the if-block. However, in most other places you do not check of the returned chain_id is 0 and add the command line option unconditionally. I think it is ok to do the same here.

bye,
Sumit

Reply:
Fixed, thank you.